test/e2e/functional/tests/e2e.go

```diff
@@ -1085,70 +951,95 @@ var _ = Describe("metallb", func() {
-`))
+  `)
 
 			By("Deleting the Configmap")
 			err = testclient.Client.ConfigMaps(OperatorNameSpace).Delete(context.Background(), consts.MetalLBConfigMapName, metav1.DeleteOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() (string, error) {
-				var err error
-				configMap, err = testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
-				if err != nil {
-					return "", err
-				}
-				return configMap.Data[consts.MetalLBConfigMapName], err
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML(`address-pools:
+			AssertConfigMapMatchYAML(`address-pools:
 - name: addresspool1
   protocol: layer2
   addresses:
   - 1.1.1.1-1.1.1.100
-`))
+`)
 
 			By("Checking AddressPool resource is deleted and ConfigMap is cleared")
 			err = testclient.Client.Delete(context.Background(), addressPool)
 			Expect(err).ToNot(HaveOccurred())
 
-			Eventually(func() string {
-				configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return configmap.Data[consts.MetalLBConfigMapName]
-			}, metallbutils.Timeout, metallbutils.Interval).Should(MatchYAML("{}"))
+			AssertConfigMapMatchYAML("{}")
 		})
 	})
 })
+
+func IsValidationWebhookEnabled() bool {
+	deploy, err := testclient.Client.Deployments(OperatorNameSpace).Get(context.Background(), consts.MetalLBOperatorDeploymentName, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	isValidationWebhookEnabled := false
+	for _, container := range deploy.Spec.Template.Spec.Containers {
+		if container.Name == "manager" {
+			for _, env := range container.Env {
+				if env.Name == "ENABLE_OPERATOR_WEBHOOK" {
+					if env.Value == "true" {
+						isValidationWebhookEnabled = true
+					}
+				}
+			}
+		}
+	}
+
+	return isValidationWebhookEnabled
+}
+
+func IsValidationWebhookRunning(validationWebhookName string) bool {
+	// Can't just check the ValidatingWebhookConfiguration name as it's changing between different deployment methods.
+	// Need to check the webhook name in the webhooks definition of the ValidatingWebhookConfiguration.
+	validateCfgList := &admv1.ValidatingWebhookConfigurationList{}
+	err := testclient.Client.List(context.TODO(), validateCfgList, &goclient.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	isValidationWebhookRunning := false
+	for _, validateCfg := range validateCfgList.Items {
+		for _, webhook := range validateCfg.Webhooks {
+			if webhook.Name == validationWebhookName {
+				isValidationWebhookRunning = true
+			}
+		}
+	}
+
+	return isValidationWebhookRunning
+}
+
+func AssertConfigMapMatchYAML(yaml string) {
+	Eventually(func() (string, error) {
+		configmap, err := testclient.Client.ConfigMaps(OperatorNameSpace).Get(context.Background(), consts.MetalLBConfigMapName, metav1.GetOptions{})
+		if err != nil {
+			return "", err
```

Comment:
the returned err is never used

Reply:
good point. I added a check for the error - the second returned value

Reply:
i'm not sure about returning the error instead of just check it and fail. if there is an error it means the configmap is not found, something is broken. i think in such state the test should just fail instead of keep looking for the configmap in a loop and spend time.

Reply:
You're absolutely right. fixed.